m_selection.c

```diff
@@ -0,0 +1,199 @@
+        if (fitQunExit[2]>=cut2) {return true;}
+        return false;
+    }
+};
+double selector_2D::cut, selector_2D::cut2, selector_2D::fitQunDist, selector_2D::fitQunLikelihood;
+double selector_2D::fitQunEntrance[3], selector_2D::fitQunExit[3];
+
+class selector_1D {
+public:
+    static double cut, fitQunDist, fitQunLikelihood;
+    static double fitQunEntrance[3], fitQunExit[3];
+    selector_1D(const char* pname, double start, double end, double interval) {
+        int bin = (end-start)/interval;
+        TH1D* entrance_diff_mean = new TH1D("entrance_diff_mean","Mean of entrance difference;cutoff;[cm]",bin,start,end);
+        TH1D* entrance_diff_sd = new TH1D("entrance_diff_sd","SD of entrance difference;cutoff;[cm]",bin,start,end);
+        TH1D* exit_diff_mean = new TH1D("exit_diff_mean","Mean of exit difference;cutoff;[cm]",bin,start,end);
+        TH1D* exit_diff_sd = new TH1D("exit_diff_sd","SD of exit difference;cutoff;[cm]",bin,start,end);
+        TH1D* dist_diff_mean = new TH1D("dist_diff_mean","Mean of distance difference;cutoff;[cm]",bin,start,end);
+        TH1D* dist_diff_sd = new TH1D("dist_diff_sd","SD of distance difference;cutoff;[cm]",bin,start,end);
+        TH1D* percentage = new TH1D("percantage","Percentage of remaining;cutoff;percentage",bin,start,end);
+
+        for (cut=start;cut<end;cut+=interval) {
+            vector<double> data = GetData<selector_1D> (pname);
+            entrance_diff_mean->Fill(cut,data[0]);
+            entrance_diff_sd->Fill(cut,data[1]);
+            exit_diff_mean->Fill(cut,data[2]);
+            exit_diff_sd->Fill(cut,data[3]);
+            dist_diff_mean->Fill(cut,data[4]);
+            dist_diff_sd->Fill(cut,data[5]);
+            percentage->Fill(cut,data[6]);
+        }
+
+        TCanvas* c1 = new TCanvas();
+        gStyle->SetOptStat(0);
+        entrance_diff_mean->Draw("hist");
+        c1->SaveAs(Form("entrance_diff_mean.pdf"));
+        delete entrance_diff_mean;
+        entrance_diff_sd->Draw("hist");
+        c1->SaveAs(Form("entrance_diff_sd.pdf"));
+        delete entrance_diff_sd;
+        exit_diff_mean->Draw("hist");
+        c1->SaveAs(Form("exit_diff_mean.pdf"));
+        delete exit_diff_mean;
+        exit_diff_sd->Draw("hist");
+        c1->SaveAs(Form("exit_diff_sd.pdf"));
+        delete exit_diff_sd;
+        dist_diff_mean->Draw("hist");
+        c1->SaveAs(Form("dist_diff_mean.pdf"));
+        delete dist_diff_mean;
+        dist_diff_sd->Draw("hist");
+        c1->SaveAs(Form("dist_diff_sd.pdf"));
+        delete dist_diff_sd;
+        percentage->Draw("hist");
+        c1->SaveAs(Form("percentage.pdf"));
+        delete percentage;
+    }
+
+    static bool selector() {
+        //if (fitQunLikelihood>=cut) {return true;}
+        if (fitQunEntrance[2]==max_z) {return true;}
+        if (fitQunExit[2]==-max_z) {return true;}
+        if (fitQunDist<=cut) {return true;}
```

Comment:
I guess you need to comment and uncomment these lines every time when you change the cut variables? Try to make this function to take an argument to identity the cut conditions.